apps/design/backend/src/convert_ms_results.ts

```diff
@@ -0,0 +1,249 @@
+import { Election, safeParseInt, Tabulation } from '@votingworks/types';
+import { parse } from 'csv-parse/sync';
+import { stringify } from 'csv-stringify/sync';
+import {
+  assert,
+  assertDefined,
+  deepEqual,
+  err,
+  find,
+  groupBy,
+  iter,
+  ok,
+  Result,
+  throwIllegalValue,
+} from '@votingworks/basics';
+import { MS_ID_SEPARATOR } from './convert_ms_election';
+
+const ALL_PRECINCTS_TALLY_REPORT_COLUMNS = [
+  { name: 'Precinct', key: 'precinct' },
+  { name: 'Precinct ID', key: 'precinctId' },
+  { name: 'Contest', key: 'contest' },
+  { name: 'Contest ID', key: 'contestId' },
+  { name: 'Selection', key: 'selection' },
+  { name: 'Selection ID', key: 'selectionId' },
+  { name: 'Total Votes', key: 'totalVotes' },
+] as const;
+
+export type AllPrecinctsTallyReportRow = {
+  [K in (typeof ALL_PRECINCTS_TALLY_REPORT_COLUMNS)[number]['key']]: string;
+};
+
+export const SEMS_RESULTS_COLUMNS = [
+  'countyId',
+  'precinctId',
+  'contestId',
+  'contestTitle',
+  'partyId',
+  'partyLabel',
+  /**
+   * Either a candidate/option ID or a code:
+   * - 0 indicates write-ins
+   * - 1 indicates overvotes
+   * - 2 indicates undervotes
+   */
+  'candidateId',
+  'candidateName',
+  'candidatePartyId',
+  'candidatePartyLabel',
+  'voteCount',
+] as const;
+
+export type SemsResultsRow = {
+  [K in (typeof SEMS_RESULTS_COLUMNS)[number]]: string;
+};
+
+const NONPARTISAN_PARTY_ID = '0';
+const NONPARTISAN_PARTY_LABEL = 'NP';
+
+export type ConvertMsResultsError =
+  | 'invalid-headers'
+  | 'report-precincts-mismatch'
+  | 'report-contests-mismatch';
+
+export function convertMsResults(
```

Comment:
Non-blocking, just noting: seems like this could be fairly hefty for a synchronous fn, looking at the logic below - maybe not an issue for the datasets we expect to see (and the low VxDesign usage), but worth keeping in mind